packages/posthog/ios/Plugin/PosthogPlugin.swift

```diff
@@ -181,6 +182,15 @@ public class PosthogPlugin: CAPPlugin, CAPBridgedPlugin {
         call.resolve()
     }
 
+    @objc func getSessionId(_ call: CAPPluginCall) {
+        let result = implementation?.getSessionId()
+        if let result = result?.toJSObject() as? JSObject {
+            self.resolveCall(call, result)
+        } else {
+            call.resolve(["sessionId": nil])
+        }
```

Comment:
```suggestion
        if let result = result?.toJSObject() as? JSObject {
            self.resolveCall(call, result)
        }
```

We assume that a result will always be available.